### PR TITLE
Add status reason and type constants for UDN resource support

### PR DESCRIPTION
**Pull request by sbahar619**
##### Short description:
This pull request adds new status reason and type constants to support the User Defined Network (UDN) resource, enhancing the clarity and functionality of network-related conditions.
##### What this PR does / why we need it:
This PR adds necessary constants to support future utility functions for User Defined Network resources.
## Summary by CodeRabbit

- **New Features**
	- Enhanced resource management with new conditions for tracking network resources.
	- Introduced a new API group for network addons in Kubernetes.
- **Improvements**
	- Updated constructor for the Resource class to allow for more parameters, increasing flexibility.
	- Enhanced `get` method functionality in Resource and NamespacedResource classes to support new conditions and types.

### CODE REVIEW

**Comment by redhat-qe-bot1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: ocp_resources/resource.py [hunk 1]
```diff
@@ -253,6 +253,11 @@ class Phase:
         class Reason:
             ALL_REQUIREMENTS_MET: str = "AllRequirementsMet"
             INSTALL_SUCCEEDED: str = "InstallSucceeded"
+            NETWORK_ATTACHMENT_DEFINITION_READY: str = "NetworkAttachmentDefinitionReady"
+            SYNC_ERROR: str = "SyncError"
```
[the remark this answers is not in the document]
Comment (sbahar619) [reply ocp_resources/resource.py:257]: It will be used in a future PR.

**Review by yossisegev — APPROVED** [review]
/lgtm

**Review by myakove — APPROVED** [review]

**Pull request merged**